include/kernel/sporadic.h

```diff
@@ -141,14 +141,10 @@ void refill_update(sched_context_t *sc, ticks_t new_period, ticks_t new_budget,
 
 
 /* Charge `usage` to the current scheduling context.
- * This function should only be called only when charging `used` will deplete
- * the head refill, resulting in refill_sufficient failing.
  *
  * @param usage the amount of time to charge.
- * @param capacity the value returned by refill_capacity. At most call sites this
```

Comment:
this comment is exactly why I passed capacity - to avoid recalculating it all the time

Reply:
The only callsite I can find where the capacity needed to be calculated already anyway was in `checkBudget` where the case where it was needed was when an SC had exhausted all of its budget when attempting a syscall. I recall this being requested from verification to reduce complexity.

Reply:
ok. It's pretty important to track why changes like this were made.

Reply:
It seems like `capacity == 0` was also used by `handleYield` to ensure that the full timeslice is consumed instead of just what has been consumed so far?